src/anomalib/metrics/collection.py

```diff
@@ -3,16 +3,22 @@
 # Copyright (C) 2022-2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
+import logging
+
 from torchmetrics import MetricCollection
 
+logger = logging.getLogger(__name__)
+
 
 class AnomalibMetricCollection(MetricCollection):
     """Extends the MetricCollection class for use in the Anomalib pipeline."""
 
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
+        self._saturation_config: dict
```

Comment:
`self._saturation_config` is defined twice here.

Also, I was wondering if it is possible to somehow exclude self._saturation_config from AnomalibMetricCollection? As you mention in the codebase, it is quite specific to a single dataset

Reply:
ah yeah, it seems we don't necessarily need the `self._saturation_config` AnomalibMetricCollection. But maybe we still need the `def set_saturation_config` method in the AnomalibMetricCollection to pass the config/path to the metric, i.e., SPRO and AUSPRO

Reply:
I too think that maybe we should explore the idea of moving this to be only sPRO specific.

Reply:
I have removed `self._saturation_config`, but moving the saturation_config completely to SPRO is not explored yet.  Maybe it depends on this comment? https://github.com/openvinotoolkit/anomalib/pull/1686#discussion_r1480484061

This code is still in the metric collection
https://github.com/openvinotoolkit/anomalib/blob/2cdeb82eb79348a23325fb3d873862775366faa2/src/anomalib/metrics/collection.py#L28-L37

Reply:
Now I have managed to remove the saturation_config related code from the metric collection in this commit https://github.com/openvinotoolkit/anomalib/pull/1686/commits/d9a233340061c70334f726869e4189d33e01c061

I will explain more in this comment https://github.com/openvinotoolkit/anomalib/pull/1686#discussion_r1478746532